lib/ansible/modules/identity/ipa/ipa_host.py

```diff
@@ -42,6 +42,12 @@
     - If an empty list is passed all assigned MAC addresses will be removed.
     - MAC addresses that are already assigned but not passed will be removed.
     aliases: ["macaddress"]
+  userpassword:
+    description:
+    - Set the one time password (OTP).
```

Comment:
This question is probably because I haven't used this function of FreeIPA yet, so if you could just fill me in a bit, I would really appreciate it...

According to the FreeIPA API for this, the `userpassword` field is the "Password used in bulk enrollment"... so how does this relate to OTP?  Is this saying that when FreeIPA is configured to use OTP, one would need to set this option here?

Reply:
In FreeIPA, users and hosts (machine password) have a password. Both use the field 'userpassword'. The machine password is stored in the /etc/krb5.keytab file and allows the machine to communicate securely with FreeIPA components. A machine can be join in the FreeIPA domain on the client side with the tool 'ipa-client-install'. Normally, an administrative account must be specified interactively. You can avoid this by creating the new machine with the ansible module 'ipa_host' and setting the machine password to a known value. Using the command line switch '--password' of 'ipa-client-install' you can specify this password and automatically join the machine to the FreeIPA domain. We also do this with ansible. In FreeIPA, the machine password option is called 'One-Time-Password' because this password is only valid for joining and is then changed. This has nothing to do with the familiar OTP two-factor authentication for users.

Reply:
Hi,
you can find more information in this [chapter](https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/7/html/linux_domain_identity_authentication_and_policy_guide/client-kickstart)
With this option, the admin creates a password that can be used during ipa-client-install --password=$value, and will be valid only for the first authentication. This means that after ipa-client-install, the password cannot be used any more.
The wording may be confusing but it does not relate at all to OTP authentication that can be performed by the user as described [there](https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/7/html/linux_domain_identity_authentication_and_policy_guide/otp)

Reply:
OKay, *got* it! This last sentence made all the sense in the world to me: 

> This has nothing to do with the familiar OTP two-factor authentication for users.

👍